x-pack/plugins/endpoint/server/routes/resolver/lifecycle.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { schema } from '@kbn/config-schema';
+import { RequestHandler, Logger } from 'kibana/server';
+import { extractParentEntityID } from './utils/normalize';
+import { LifecycleQuery } from './queries/lifecycle';
+import { ResolverEvent } from '../../../common/types';
+
+interface LifecycleQueryParams {
+  ancestors: number;
+  legacyEndpointID?: string;
```

Comment:
❔ That's really optional?

Reply:
Yeah, the reason this is optional is because there are two different types of IDs

**Legacy**
A legacy Entity ID is made up of the `agent.id` and `unique_pid` fields. The client will need to identify if it's looking at an evrelay (phase 0) event and use those fields when making requests to the backend. The request would be `/resolver/{id}?legacyEndpointID=<some uuid>`and the `{id}` would be the `unique_pid`.

**Elastic Endpoint**
When interacting with the new form of data the client doesn't need the `legacyEndpointID` because it's already a part of the entityID in the new type of event. So for the same request the client would just hit `resolver/{id}` and the `{id}` would be entityID stored in the event's `process.entity_id` field.

Reply:
Yep, what Jon said, the `unique_pid` stuff in our original implementation is specific to an individual endpoint, so to uniquely identify, we need a tuple of `( agent.id, endgame.unique_pid)` to actually uniquely identify -- for our new stuff, `process.entity_id` is considered globally unique already, so no need to pass anything else.

Reply:
^ Some of this should maybe be in a comment

Reply:
@bkimmel I added the explanation to as a comment.

Reply:
Good explanation in the comment